lib/Onyx.js

```diff
@@ -908,67 +955,47 @@ function multiSet(data) {
  */
-function applyMerge(key, data) {
-    const mergeValues = mergeQueue[key];
-    if (_.isArray(data) || _.every(mergeValues, _.isArray)) {
-        // Array values will always just concatenate
-        // more items onto the end of the array
-        return _.reduce(mergeValues, (modifiedData, mergeValue) => [
-            ...modifiedData,
-            ...mergeValue,
-        ], data || []);
+function applyMerge(existingValue, changes) {
+    if (_.isUndefined(existingValue) || _.isNull(existingValue)) {
+        return changes;
+    }
+
+    if (_.isArray(existingValue) || _.isArray(changes)) {
+        return changes;
     }
 
-    if (_.isObject(data) || _.every(mergeValues, _.isObject)) {
-        // Object values are merged one after the other
-        return _.reduce(mergeValues, (modifiedData, mergeValue) => {
-            // lodash adds a small overhead so we don't use it here
-            // eslint-disable-next-line prefer-object-spread, rulesdir/prefer-underscore-method
-            const newData = Object.assign({}, fastMerge(modifiedData, mergeValue));
-
-            // We will also delete any object keys that are undefined or null.
-            // Deleting keys is not supported by AsyncStorage so we do it this way.
-            // Remove all first level keys that are explicitly set to null.
-            return _.omit(newData, (value, finalObjectKey) => _.isNull(mergeValue[finalObjectKey]));
-        }, data || {});
+    if (_.isObject(existingValue) || _.isObject(changes)) {
+        // lodash adds a small overhead so we don't use it here
+        // eslint-disable-next-line prefer-object-spread, rulesdir/prefer-underscore-method
+        const newData = fastMerge(existingValue, changes);
+
+        // We will also delete any object keys that are undefined or null.
+        // Remove all first level keys that are explicitly set to null.
+        return _.omit(newData, value => _.isNull(value));
     }
 
     // If we have anything else we can't merge it so we'll
     // simply return the last value that was queued
-    return _.last(mergeValues);
+    return changes;
 }
 
 /**
  * Merge a new value into an existing value at a key.
  *
  * The types of values that can be merged are `Object` and `Array`. To set another type of value use `Onyx.set()`. Merge
- * behavior uses lodash/merge under the hood for `Object` and simple concatenation for `Array`. However, it's important
- * to note that if you have an array value property on an `Object` that the default behavior of lodash/merge is not to
- * concatenate. See here: https://github.com/lodash/lodash/issues/2872
+ * behavior uses lodash/merge under the hood for `Object` and
+ * for `Array`'s we simply replace the current value with the new one.
  *
  * Calls to `Onyx.merge()` are batched so that any calls performed in a single tick will stack in a queue and get
  * applied in the order they were called. Note: `Onyx.set()` calls do not work this way so use caution when mixing
```

Comment:
Is this comment still true?

Reply:
hmm, well it differs between web and native now. We could just remove the part where it says, "uses lodash/merge under the hood". It's not important for using `merge` anyway